src/java/boa/aggregators/MeanAggregator.java

```diff
@@ -28,9 +28,17 @@
 
 	public void count(final String metadata) {
 		if (metadata == null)
+		{
 			this.count++;
+		}			
 		else
-			this.count += Long.parseLong(metadata);
+		{
+			try {
+				this.count += Long.parseLong(metadata);
+			} catch (NumberFormatException e) {
+				this.count += (long)Double.parseDouble(metadata);
```

Comment:
If the values are: 2.9, 3.9, 2.9 then the mean *should be* 3.23333333.

With this solution, you instead get 2+3+2=7/3 = 2.33333333